### ISSUE TITLE
Allow duplicate folder name creation and use

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#50]
**Originally by _flabrie@users.sourceforge.net_ on 2010-03-04 18:39:31 +0100**

___
Since YAM 2.4 now support hierachical folders, it would be nice to allow duplicate folder name. Since I could now classify my folder in category, I would like to organize them logically, like:

```
+ Mac OS X
| + News
| + Maling Lists
| + ...
+ AmigaOS
+ News
+ Mailing lists
+ ...
```
Right now it's impossible. It would be necessary to generate an hidden unique ID or use the full path reference to associate filter move function to folder instead of just the folder name.

By the way, this reference to an hidden ID would be far better and would be more flexible than name reference. When a user will change a folder name, it would not be necessary to change all filter associated to this folder.

**Comment by jens-maus** [comment]
**Originally on 2013-01-19 21:47:00 +0100**

___
Makes sense and is in line with what wew need for full IMAP usage later on anyway. We should really allow to have folder with equal names, thus differentiate between display name and folder path.

**Comment by tboeckel** [comment]
**Originally on 2013-11-15 10:14:33 +0100**

___
In (04deb87):
 * YAM_FO.c, Config.c, FolderList.c, misc: each folder now gets a unique ID assigned. In the future everything will be changed to reference folders by this ID only instead of referencing them by name, path or type. This will eventually make it possible to have equally named folders in different paths. This refs #400 and also refs #50.

**Comment by jens-maus** [comment]
**Originally on 2014-05-12 09:23:39 +0200**

___
Now that we have the functionality in YAM to refer to a folder based on a ID we should definitely  implement this for 2.10

**Comment by tboeckel** [comment]
**Originally on 2014-08-18 08:50:53 +0200**

___
In (9db5b47):
* mui/FolderEditWindow.c: removed the restriction of unique folder names. This makes it possible to have equally named folders. This closes #50.
